libcore/src/Simulation.cpp

```diff
@@ -214,93 +215,55 @@ double Simulation::RunStandardSimulation(double maxSimTime)
-    const std::vector<Pedestrian *> & allPeds = _building->GetAllPedestrians();
-    const std::map<int, Goal *> & goals       = _building->GetAllGoals();
-    auto allRooms                             = _building->GetAllRooms();
-
-#pragma omp parallel for shared(pedsToRemove, allRooms)
-    for(size_t p = 0; p < allPeds.size(); ++p) {
-        auto ped       = allPeds[p];
-        Room * room    = _building->GetRoom(ped->GetRoomID());
-        SubRoom * sub0 = room->GetSubRoom(ped->GetSubRoomID());
-
-        //set the new room if needed
-        if((ped->GetFinalDestination() == FINAL_DEST_OUT) &&
-           (room->GetCaption() == "outside")) { //TODO Hier aendern fuer inside goals?
-#pragma omp critical(Simulation_Update_pedsToRemove)
-            pedsToRemove.insert(ped);
-        } else if(
-            (ped->GetFinalDestination() != FINAL_DEST_OUT) &&
-            (goals.at(ped->GetFinalDestination())->Contains(ped->GetPos())) &&
-            (goals.at(ped->GetFinalDestination())->GetIsFinalGoal())) {
-#pragma omp critical(Simulation_Update_pedsToRemove)
-            pedsToRemove.insert(ped);
-        }
-
-        // reposition in the case the pedestrians "accidentally left the room" not via the intended exit.
-        // That may happen if the forces are too high for instance
-        // the ped is removed from the simulation, if it could not be reassigned
-        else if(!sub0->IsInSubRoom(ped)) {
-            bool assigned = false;
-            std::function<void(const Pedestrian &)> f =
-                std::bind(&Simulation::UpdateFlowAtDoors, this, std::placeholders::_1);
-            assigned = ped->Relocate(f);
-            //this will delete agents, that are pushed outside (maybe even if inside obstacles??)
-
-            if(!assigned) {
-#pragma omp critical(Simulation_Update_pedsToRemove)
-                pedsToRemove.insert(ped); //the agent left the old room
-                                          //actualize the eggress time for that room
-#pragma omp critical(SetEgressTime)
-                allRooms.at(ped->GetRoomID())->SetEgressTime(Pedestrian::GetGlobalTime());
-            }
-        }
+    //TODO update egress time of room
+    //TODO discuss program flow of main loop
+    // ------------------------------------------------
+    auto peds = _building->GetAllPedestrians();
+
+    auto [pedsChangedRoom, pedsNotRelocated] =
+        SimulationHelper::UpdatePedestrianLocation(*_building, peds);
+
+    // TODO not needed at the moment, as we do not have inside final goals yet
+    auto pedsAtFinalGoal = SimulationHelper::FindPedsReachedFinalGoal(*_building, peds);
+    _pedsToRemove.insert(pedsAtFinalGoal.begin(), pedsAtFinalGoal.end());
+
+    auto pedsOutside = SimulationHelper::FindPedestriansOutside(*_building, pedsNotRelocated);
+    _pedsToRemove.insert(pedsNotRelocated.begin(), pedsNotRelocated.end());
+    _pedsToRemove.insert(pedsOutside.begin(), pedsOutside.end());
+
+    //TODO see if this can be merged
```

Comment:
Can remove some of the TODO comments before merging?

Reply:
Removed